controllers/virtualmachine/virtualmachine_controller.go

```diff
@@ -87,12 +89,31 @@ func AddToManager(ctx *context.ControllerManagerContext, mgr manager.Manager) er
 			handler.EnqueueRequestsFromMapFunc(csBindingToVMMapperFn(ctx, r.Client)))
 	}
 
-	if !lib.IsNamespacedVMClassFSSEnabled() {
-		builder = builder.Watches(&source.Kind{Type: &vmopv1.VirtualMachineClassBinding{}},
-			handler.EnqueueRequestsFromMapFunc(classBindingToVMMapperFn(ctx, r.Client)))
+	// Predicate functions to determine when a reconcile should be triggered
+	// Used to only watch for VMClass Create and Delete event
+	eventFilter := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return true
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return true
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return false
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return false
+		},
+	}
+	if lib.IsNamespacedVMClassFSSEnabled() {
+		builder = builder.Watches(
+			&source.Kind{Type: &vmopv1.VirtualMachineClass{}},
+			handler.EnqueueRequestsFromMapFunc(classToVMMapperFn(ctx, r.Client))).
+			WithEventFilter(eventFilter)
```

Comment:
```suggestion
			WithEventFilter(predicate.Funcs{
 		            CreateFunc: func(e event.CreateEvent) bool {
 			            return true
 		            },
 		            DeleteFunc: func(e event.DeleteEvent) bool {
 			            return true
 		            },
 		            UpdateFunc: func(e event.UpdateEvent) bool {
 			            return false
 		            },
 		            GenericFunc: func(e event.GenericEvent) bool {
 			            return false
 		            },
 	                 })
```
nit: since we do not use the eventFilter any place else. We should eventually consider moving any and all predicates to the new package `pkg/util/kube/predicates.go` file